client/app/components/BottomTabs.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+import { withNavigation } from 'react-navigation';
+
+import connectToRedux from '../redux/lib/connectToRedux';
+import sp from '../redux/lib/stateProperties';
+
+import styles, { colors } from '../config/styles';
+import SimpleLineIcons from '@expo/vector-icons/SimpleLineIcons';
+import MaterialCommunityIcons from '@expo/vector-icons/MaterialCommunityIcons';
+import MaterialIcons from '@expo/vector-icons/MaterialIcons';
+
+class BottomTabs extends Component {
+
+    constructor(props) {
+        super(props);
+        console.log(this.props.navigation)
+    }
+
+    render() {
+        return (
+            <View style={tabStyles.container}>
+                <View style={{
+                    ...styles.container.backgroundColorPrimary, 
+                    ...tabStyles.bottomTabs,
+                }}>
+                    <TouchableOpacity onPress={
```

Comment:
Could this possibly be refactored to avoid the copypasta-y nature of the `TouchableOpacity` components? For example having some type of array with the info we need to pass in?

```javascript
tabButtons = [
  {
    viewName: 'FriendsView',
    IconComponent: SimpleLineIcons,
    iconName: 'people',
  },
  {
    viewName: 'DiningHallsView',
    IconComponent: MaterialCommunityIcons,
    iconName: 'food',
  },
  {
    viewName: 'FavoritesView',
    IconComponent: MaterialIcons,
    iconName: 'favorite',
  }
]
```

And then in the `render` function

```jsx
<View>
  {this.tabButtons.map(tabButton => {
    return (
      <TouchableOpacity onPress={() => this.props.navigation.navigate(tabButton.viewName)}>
        <tabButton.IconComponent name={tabButton.iconName} size={25}, color={colors.secondary}
        />
      </TouchableOpacity>
    )
  })}
</View>
```

Or something like that/however you think would be best?

Reply:
Ooh nice